swift/ql/src/queries/Summary/QuerySinks.ql

```diff
@@ -0,0 +1,84 @@
+/**
+ * @name Query Sinks
+ * @description List all query sinks found in the database. Query sinks are
+ *              potential results depending on what data flows to them and
+ *              other context.
+ * @kind problem
+ * @problem.severity info
+ * @id swift/summary/query-sinks
+ * @tags summary
+ */
+
+/*
+ * Most queries compute data flow to one of the following sinks:
+ *  - custom per-query sinks (listed by this query, `swift/summary/query-sinks`).
+ *  - regular expression evaluation (see `swift/summary/regex-evals`).
+ */
+
+import swift
+import codeql.swift.dataflow.DataFlow
+import codeql.swift.security.PathInjectionQuery
+import codeql.swift.security.UnsafeWebViewFetchQuery
+import codeql.swift.security.SqlInjectionQuery
+import codeql.swift.security.UnsafeJsEvalQuery
+import codeql.swift.security.UncontrolledFormatStringQuery
+import codeql.swift.security.StringLengthConflationQuery
+import codeql.swift.security.ConstantPasswordQuery
+import codeql.swift.security.CleartextStorageDatabaseQuery
+import codeql.swift.security.CleartextTransmissionQuery
+import codeql.swift.security.CleartextLoggingQuery
+import codeql.swift.security.CleartextStoragePreferencesQuery
+import codeql.swift.security.HardcodedEncryptionKeyQuery
+import codeql.swift.security.ECBEncryptionQuery
+import codeql.swift.security.WeakSensitiveDataHashingQuery
+import codeql.swift.security.XXEQuery
+import codeql.swift.security.InsecureTLSQuery
+import codeql.swift.security.ConstantSaltQuery
+import codeql.swift.security.InsufficientHashIterationsQuery
+import codeql.swift.security.PredicateInjectionQuery
+import codeql.swift.security.StaticInitializationVectorQuery
+
+string queryForSink(DataFlow::Node sink) {
+  PathInjectionConfig::isSink(sink) and result = "swift/path-injection"
```

Comment:
This _almost_ makes me wish every dataflow module that fully defined a query defined a predicate like:
```ql
module PathInjectionConfig implements DataFlow::ConfigSig {
  ...
  
  string queryId() {
    result = "swift/path-injection"
  }
}
```
but that's probably too much effort to put into making this table-like diagnostics query.

Reply:
Yeah, that would be nice.  Regardless there will be a small maintenance burden adding new queries to this list, but I think it's worth it for the data we get.  At some point `queryForSink` may move out into a shared `.qll` so that we can write other queries using it.